### PR TITLE
Add support for hexadecimal numbers

### PR DESCRIPTION
**Pull request by Gabriel439**
Fixes https://github.com/dhall-lang/dhall-lang/issues/858

### CODE REVIEW

**Comment by Gabriel439** [comment]
Matching change to the Haskell implementation: https://github.com/dhall-lang/dhall-haskell/pull/1607

File: standard/dhall.abnf [hunk 1]
```diff
@@ -486,7 +486,10 @@ double-literal =
     ; "NaN"
     / NaN
 
-natural-literal = 1*DIGIT
+natural-literal =
+    ; Hexadecimal
```
Comment (sjakobi) [inline standard/dhall.abnf]: To explain what that `%x78` is:

```suggestion
    ; Hexadecimal with "0x" prefix
```

**Comment by SiriusStarr** [comment]
[Cheatsheet](https://github.com/dhall-lang/dhall-lang/blob/master/docs/howtos/Cheatsheet.md) (line 40-ish, since Github doesn't like linking to lines in markdown files) should probably be updated so people are aware of this as an option.  As well as [Built-in Types](https://github.com/dhall-lang/dhall-lang/blob/master/docs/references/Built-in-types.md) (line 234 ish).  Unsure in both of those if you think mentioning that `Integer`s can be written as hex is a good addition as well.

File: docs/howtos/Cheatsheet.md [hunk 2]
```diff
@@ -42,6 +42,8 @@
     ```dhall
     0, 1, 2, … : Natural
 
+    0x0, 0x1, …, 0xE, 0xF, 0x10, … : Natural -- Hexadecimal notation
```
Comment (SiriusStarr) [inline docs/howtos/Cheatsheet.md]: Since the grammar allows for lowercase hex literals, should the documentation include a lowercase example as well?

Comment (Gabriel439) [reply docs/howtos/Cheatsheet.md]: @SiriusStarr: I will probably only add the lowercase examples to the built-ins document.  My reasoning is that the Reference section is optimized for comprehensiveness, unlike the How-To section.

Also, a user working off the Cheatsheet has an easy to way to confirm whether or not lowercase hexadecimal literals work: just try it!  Once the user is aware that the feature exists at all they can play with the feature to test its limits.

Comment (SiriusStarr) [reply docs/howtos/Cheatsheet.md]: Yep, sounds good to me; whatever you think is best.  Just figured it should show up somewhere in the docs (if only so people don't try it and then report it as a bug, thinking it's unintentional, haha).

**Pull request merged**